src/main/java/com/teamabnormals/abnormals_core/core/events/CompatEvents.java

```diff
@@ -33,22 +34,20 @@ public static void onRightClickEntity(PlayerInteractEvent.EntityInteract event)
 		ItemStack stack = event.getItemStack();
 		if (target instanceof IAgeableEntity && stack.getItem() == Items.POISONOUS_POTATO && ACConfig.ValuesHolder.isPoisonPotatoCompatEnabled() && ModList.get().isLoaded("quark")) {
 			PlayerEntity player = event.getPlayer();
-			CompoundNBT persistantData = target.getPersistentData();
-			if (((IAgeableEntity) target).getGrowingAge() < 0 && !persistantData.getBoolean(POISON_TAG)) {
-				if (!event.getWorld().isRemote) {
-					if (target.world.rand.nextDouble() < ACConfig.ValuesHolder.poisonEffectChance()) {
-						target.playSound(SoundEvents.ENTITY_GENERIC_EAT, 0.5f, 0.25f);
-						persistantData.putBoolean(POISON_TAG, true);
-						if (ACConfig.ValuesHolder.shouldPoisonEntity()) {
-							((LivingEntity) target).addPotionEffect(new EffectInstance(Effects.POISON, 200));
-						}
-					} else {
-						target.playSound(SoundEvents.ENTITY_GENERIC_EAT, 0.5f, 0.5f + target.world.rand.nextFloat() / 2);
+			CompoundNBT persistentData = target.getPersistentData();
+			if (((IAgeableEntity) target).canAge(true) && !persistentData.getBoolean(POISON_TAG)) {
+				if (target.world.rand.nextDouble() < ACConfig.ValuesHolder.poisonEffectChance()) {
+					target.playSound(SoundEvents.ENTITY_GENERIC_EAT, 0.5f, 0.25f);
+					persistentData.putBoolean(POISON_TAG, true);
+					if (ACConfig.ValuesHolder.shouldPoisonEntity()) {
+						((LivingEntity) target).addPotionEffect(new EffectInstance(Effects.POISON, 200));
 					}
 				} else {
-					player.swingArm(event.getHand());
+					target.playSound(SoundEvents.ENTITY_GENERIC_EAT, 0.5f, 0.5f + target.world.rand.nextFloat() / 2);
 				}
 				if (!player.isCreative()) stack.shrink(1);
+				event.setCancellationResult(ActionResultType.func_233537_a_(event.getWorld().isRemote()));
+				event.setCanceled(true);
```

Comment:
Is `event.setCanceled(true);` neccessary?

Reply:
I think so. The doc comment for `getCancellationResult` implies that, anyway:
```
/**                                                                                                                                                 
 * Set the EnumActionResult that will be returned to vanilla if the event is cancelled, instead of calling the relevant                             
 * method of the event.                                                                                                                             
 * Note that this only has an effect on {@link RightClickBlock}, {@link RightClickItem}, {@link EntityInteract}, and {@link EntityInteractSpecific}.
 */                                                                                                                                                 
```

Reply:
I just looked at the code for it and it does need it.